java/code/src/com/redhat/rhn/domain/user/UserFactory.java

```diff
@@ -708,7 +708,7 @@ public void setUserServerPreferenceValue(User user,
      */
     public List<User> findAllUsers(Optional<Org> inOrg) {
         Map<String, Object> params = new HashMap<>();
-        int orgId = Opt.fold(inOrg, () -> -1, org -> org.getId().intValue());
+        Integer orgId = Opt.fold(inOrg, () -> null, org -> org.getId().intValue());
         params.put("org_id", orgId);
```

Comment:
I wonder of the problem is, that we need to put `int` or `null` here. Seems that `Integer` is not working but we need `null` for the other check.

Reply:
Exact same error happened with `Long` type, which was the original one passed to params. It seems like its something around `:org_id IS NULL`, which is weird because the query works well if I try it directly on psql. Could this be because we are using the parameter `:org_id` twice on the same expression?

Reply:
After asking google it seems that this kind of statement is not working with HQL.
So I think you need to statements and depending of orgId use the one or the other.

Reply:
My love for hibernate is not growing. Just pushed additional changes, with the two statements.